internal/elasticsearch/enrich/models.go

```diff
@@ -0,0 +1,45 @@
+package enrich
+
+import (
+	"context"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/models"
+	"github.com/elastic/terraform-provider-elasticstack/internal/utils"
+	"github.com/hashicorp/terraform-plugin-framework-jsontypes/jsontypes"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+type EnrichPolicyData struct {
+	Id                      types.String         `tfsdk:"id"`
+	ElasticsearchConnection types.List           `tfsdk:"elasticsearch_connection"`
+	Name                    types.String         `tfsdk:"name"`
+	PolicyType              types.String         `tfsdk:"policy_type"`
+	Indices                 types.List           `tfsdk:"indices"`
+	MatchField              types.String         `tfsdk:"match_field"`
+	EnrichFields            types.List           `tfsdk:"enrich_fields"`
+	Query                   jsontypes.Normalized `tfsdk:"query"`
+	Execute                 types.Bool           `tfsdk:"execute"`
+}
```

Comment:
```suggestion
type EnrichPolicyData struct {
	Id                      types.String         `tfsdk:"id"`
	ElasticsearchConnection types.List           `tfsdk:"elasticsearch_connection"`
	Name                    types.String         `tfsdk:"name"`
	PolicyType              types.String         `tfsdk:"policy_type"`
	Indices                 types.List           `tfsdk:"indices"`
	MatchField              types.String         `tfsdk:"match_field"`
	EnrichFields            types.List           `tfsdk:"enrich_fields"`
	Query                   jsontypes.Normalized `tfsdk:"query"`
}

type EnrichPolicyDataWithExecute struct {
	EnrichPolicyData
	Execute                 types.Bool           `tfsdk:"execute"`
}
```

The data source doesn't define the `execute` attribute in it's schema. Use the `EnrichPolicyData` model in the data source and `EnrichPolicyDataWithExecute` within the resource.